src/object/cli_ec.c

```diff
@@ -1699,6 +1701,17 @@ obj_ec_update_iod_size(struct obj_reasb_req *reasb_req, uint32_t iod_nr)
 		D_ASSERT(re_iods[i].iod_type == u_iods[i].iod_type);
 		u_iods[i].iod_size = re_iods[i].iod_size;
 	}
+
+	/* Set back the size if it is recovery task */
+	if (unlikely(fail_info != NULL)) {
+		for (i = 0; i < fail_info->efi_recov_ntasks; i++) {
+			if (fail_info->efi_recov_tasks[i].ert_iod.iod_type !=
+							DAOS_IOD_SINGLE)
```

Comment:
just confirm, why don't do it for array here?

Reply:
I suppose it can, just be cautious here.